packages/puppeteer-core/src/util/decorators.ts

```diff
@@ -57,3 +59,26 @@ export function moveable<
   }
   return Class;
 }
+
+export function withIsolatedRealm<
+  This extends ElementHandle<Node>,
+  Args extends unknown[],
+  Ret,
+>(
+  target: (this: This, ...args: Args) => Promise<Ret>,
+  _: unknown
```

Comment:
Note this `_: unknown` is required for TypeScript to understand this function is a decorator (it needs to be a function of two arguments: the target and context [which we don't type here]).